Fixes some bugs from ShowManager Object.

devmessias opened the pull request:
Related with #483 
This PR adress two minor issues with the ShowManager object

## 1.  destroy timer behavior

ShowManager destroy_timer method destroys the timer without removing the callback related with the observer. Thus,
we lost the reference for the callback. Because of that, another timer event created after that will keep calling the
callback function with the missed reference.

## 2.  Size Window attribute
This is more trickly and usually happens on Windows. The vtk creates a window instance with a slightly different size.
Therefore, the following test can fail:
```python
size = (300, 400)
showm = ShowManager(size=size)
# code goes here
assert size[0] = showm.size[0] and size[1] == showm.size[1]
I noticed that bug running the tests of PR #437 on windows github actions.
```


This is an example which shows the bug solved by this PR. 

```python
from fury.window import ShowManager
from fury.actor import sphere
import numpy as np

centers = np.array([[1, 0,0]])
showm = ShowManager(size=(400, 400))
i1 = 0
i2 = 0

#check_bug = False  

def t1(_, __):
    global i1
    i1 += 1
    print(f'timer 1 event {i1}')

def t2(_, __):
    global i2
    i2 += 1
    print(f'timer 2 event {i2}')

id1 = showm.add_timer_callback(False, 1, t1)
print(id1, showm._timers)
id2 = showm.add_timer_callback(False, 2, t2)

print(id2, showm._timers)
showm.destroy_timer(id1)

showm.initialize()

color = np.array([0, 1, 0])
actor = sphere(centers, color)
showm.scene.add(actor)

showm.start()
#showm.render()
assert i1 == 0 # the code will fail here
```
After this PR, the above code should work whithout problems

devmessias committed:
fix: a simple method to remove a Timer from a ShowManager instance

devmessias committed:
use a dict to store the references for timers and observers

This will avoid changing the ShowManager API

devmessias committed:
fix: adding a new callback for a observer should return the related id

devmessias committed:
after this, the size attribute will come directly from vtk GetScene Method.

We should do this in that way because sometimes vtk creates a slight
different windows instance than requested

devmessias committed:
docs: docstring improvements

skoudoro reviewed (commented):
Hi @devmessias,

Looks good to me. Can you add some tests to make sure that the detected bug will not appear again?

After that, we should merge this PR. Thanks!

devmessias commented:
> Hi @devmessias,
> 
> Looks good to me. Can you add some tests to make sure that the detected bug will not appear again?
> 
> After that, we should merge this PR. Thanks!

Hi @skoudoro. I'm trying to figure out how to test this whithout using showm.start(). I couldn't find a method inside of  vtkRenderWindowInteractor  which returns the list of observers and timers. In a vtkActor instance we have Mapper().GetObservers()

devmessias committed:
tests: add and destroy timer callback

pep8speaks commented:
Hello @devmessias! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`fury/tests/test_window.py`](https://github.com/fury-gl/fury/blob/c6eb4a77ad75901d1366af668a4a71145cd56208/fury/tests/test_window.py):

> [Line 342:46](https://github.com/fury-gl/fury/blob/c6eb4a77ad75901d1366af668a4a71145cd56208/fury/tests/test_window.py#L342): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'

devmessias committed:
tests: test showm.exit method

devmessias committed:
test: remove the interactive arg

devmessias committed:
Merge branch 'master' into fix_show_manager_timer_size_behaviors

skoudoro commented on fury/tests/test_window.py:
```diff
@@ -336,6 +337,60 @@ def test_content(filename='fury.png', colors_found=(True, True)):
+    showm.scene.add(sphere)
+
+    state['first_time'] = time.time()
+    if interactive:
+        showm.start()
+        # callback 1 was not called
+        npt.assert_equal(counts_by_interval[0], 0)
+        npt.assert_equal(counts_by_interval[1] >= ms_3//ms_2, True)
```
interactive is `False` so I do not know what you are testing in this test. Can you clarify?

devmessias replied:
I need to fix that.  The reason for  this is because I seems it's impossible check the `counts_by_interval` without start the ShowManager.

skoudoro commented on fury/window.py:
```diff
@@ -512,39 +511,107 @@ def play_events_from_file(self, filename):
+            If True, the timer will repeat.
+        duration : int
+            Timer duration in milliseconds.
+        timer_callback : function
```
`callable` instead of `function`. (can be a class for example)

skoudoro commented on fury/window.py:
```diff
@@ -512,39 +511,107 @@ def play_events_from_file(self, filename):
+
+        Parameters
+        ----------
+        win_callback : function
```
`callable` instead of `function`. (can be a class for example)

skoudoro reviewed (commented):
Hi @devmessias,

This PR is almost ready to go, thank you! 

Can you address the pep8 and the comments below?

Garyfallidis commented:
@devmessias this PR needs more work. First of all, you need to explain in the description which exact bugs you are fixing. Referencing another issue is not enough. Also you have to explain a bit on the test what you are testing. Currently it is not clear.
At least in Windows I do not see the expected behavior and when I enable interactive = True I get this error:

```
Traceback (most recent call last):
  File "test_window.py", line 490, in <module>
    test_timers()
  File "test_window.py", line 391, in test_timers
    npt.assert_equal(counts_by_interval[1] >= ms_3//ms_2, True)
  File "C:\Users\elef\AppData\Local\Continuum\anaconda3\lib\site-packages\numpy\testing\_private\utils.py", line 427, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: False
 DESIRED: True
```
Also there are still some free flying comments and pep 8 issues.
Please address the issues reported here and fix the title too. The title is very generic and does not help reviewing.
Thanks in advance!

Garyfallidis commented:
Ping @devmessias!

devmessias committed:
fix docstrings: function to callable

devmessias committed:
tests: skip test_timers_count if running in a Win

The reason to do that is because is
not possible to call the exit method
inside of a timer callback.

skoudoro commented:
Hi @devmessias,

Thank you for your update! Could you also rebase to resolve those conflicts?
